extensions-contrib/azure-extensions/src/main/java/org/apache/druid/storage/azure/AzureStorage.java

```diff
@@ -95,13 +87,34 @@ public long getBlobLength(final String containerName, final String blobPath)
 
   public InputStream getBlobInputStream(final String containerName, final String blobPath)
       throws URISyntaxException, StorageException
+  {
+    return getBlobInputStream(0L, containerName, blobPath);
+  }
+
+  public InputStream getBlobInputStream(long offset, final String containerName, final String blobPath)
+      throws URISyntaxException, StorageException
   {
     CloudBlobContainer container = getCloudBlobContainer(containerName);
-    return container.getBlockBlobReference(blobPath).openInputStream();
+    return container.getBlockBlobReference(blobPath).openInputStream(offset, null, null, null, null);
   }
 
   public boolean getBlobExists(String container, String blobPath) throws URISyntaxException, StorageException
   {
     return getCloudBlobContainer(container).getBlockBlobReference(blobPath).exists();
   }
+
+  @VisibleForTesting
+  CloudBlobClient getCloudBlobClient()
+  {
+    return this.cloudBlobClient;
+  }
+
+  private CloudBlobContainer getCloudBlobContainer(final String containerName)
```

Comment:
nit: this method not just returns a container, but can create it if it doesn't exist. Maybe better to rename to something more clear such as `getOrCreateCloudBlobContainer()`.

Reply:
fixed